### ISSUE TITLE
Stopped to work after upgrade to tmux 2.3

### ISSUE THREAD

**Issue by sergei-dyshel** [tpope/vim-dispatch#192]
I'm using the plugin with gvim to run commands asynchronously in tmux panes.

After upgrading tmux from version 2.2 to 2.3, I suddenly can't run properly commands with `:Make` and `:Dispatch`. Symptoms: only some output is displayed in tmux pane, than tmux freezes and stops responding. 

I'll try to gather more details later.

**Comment by tpope** [comment]
Can you try `:set shellpipe+=\ ' (with a trailing space) and see if that changes things?

**Comment by petobens** [comment]
I was having the same issue and using `:set shellpipe+\ ` seems to fix it :) .

**Comment by elliottt** [comment]
If I manually `:set shellpipe+=\ ` everything works fine, but if I put that in my .vimrc, I still get odd behavior with `:Make`.

**Comment by tpope** [comment]
Dispatch uses `pipe-pane` to intercept tmux's output, which is a cute trick to leave color enabled. Changing `'shellpipe'` tells Dispatch you have special piping needs, disabling that behavior. So, I'm guessing something changed with `pipe-pane`.

"Odd behavior" doesn't sound good. I'm almost afraid to ask.

**Comment by AndroidOatmeal** [comment]
@tpope I am getting the same issue with `:set shellpipe+=\ ` not quite working in my vimrc. Specifically, it looks like when I set it in my vimrc and run `:Make`, the quickfix list is only populated with stdout, not stderr. For some reason it works fine when the shellpipe is manuall set in an existing vim session.

**Comment by petobens** [comment]
The problem I have when manually setting `:set shellpipe+=\ ` and using `Make` is that  I don't see any output in the split tmux pane.

**Comment by elliottt** [comment]
I was able to fix this by fully setting shellpipe in my vimrc:
https://github.com/elliottt/vim-config/blob/master/vimrc#L130

**Comment by tpope** [comment]
Oh, of course, `'shellpipe'` hasn't been initialized when the vimrc is loaded.

**Comment by tpope** [comment]
Anyways if someone could investigate if there were some changes to `pipe-pane` that would be great. I'm open to disabling by default if there's not a simple answer.

**Comment by m1foley** [comment]
If pipe-pane has been changed, it doesn't seem to be on purpose. All the recent changes to [cmd-pipe-pane.c](https://github.com/tmux/tmux/commits/master/cmd-pipe-pane.c) seem to be refactors, and no pipe-pane changes are mentioned in the [changelog](https://github.com/tmux/tmux/blob/master/CHANGES).

**Comment by m1foley** [comment]
Also, the `:set shellpipe+=\ ` workaround fixes it for me.

**Comment by toiffel** [comment]
> All the recent changes to cmd-pipe-pane.c seem to be refactors, and no pipe-pane changes are mentioned in the changelog.

Yeah, in fact there were no changes in this file between tmux 2.2 and 2.3.

Anyway, the bug is reproducible with bare tmux, vim-dispatch has nothing to do with it. Created another issue https://github.com/tmux/tmux/issues/594 in tmux repo.

**Comment by toiffel** [comment]
The fix has been pushed into tmux repo.

Please try to build tmux from master branch - now it should work just fine.

**Comment by petobens** [comment]
@toiffel thank you! It works great.

**Comment by sloria** [comment]
Confirmed--upgrading to tmux 2.4 fixed this issue.

On Mac OSX with homebrew:

```
brew upgrade tmux --HEAD
```

**Comment by dpo** [comment]
I'm using Tmux 2.8 on macOS Mojave (10.14.2) with Terminal (not iTerm2). I configured a `b:dispatch` command to compile LaTeX files. When Vim 8.1.500 is opened inside a Tmux session, issuing `Dispatch!` executes the right command, but is not async; I have to watch the entire compilation until the end... What am I missing?

I tried to set `shellpipe` as suggested in this thread, to build the master branch of Tmux, to use the vim executable from MacVim, but the issue persists.
